audittools/backing_store_sql_test.go

```diff
@@ -0,0 +1,300 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company
+// SPDX-License-Identifier: Apache-2.0
+
+package audittools
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	_ "github.com/lib/pq"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sapcc/go-api-declarations/cadf"
+
+	"github.com/sapcc/go-bits/assert"
+	"github.com/sapcc/go-bits/easypg"
+)
+
+// TestMain sets up a test database server for all tests
+func TestMain(m *testing.M) {
+	easypg.WithTestDB(m, func() int {
+		return m.Run()
+	})
+}
+
+// emptyMigration provides a minimal migration config for easypg.
+// The SQLBackingStore creates its own table via ensureTableExists(),
+// so we only need to satisfy easypg's requirement for a non-empty migration map.
+func emptyMigration() easypg.Configuration {
+	return easypg.Configuration{
+		Migrations: map[string]string{
+			"001_empty.up.sql":   "SELECT 1",
+			"001_empty.down.sql": "SELECT 1",
+		},
+	}
+}
+
+// TestSQLBackingStoreWriteAndRead tests basic write and read operations.
+func TestSQLBackingStoreWriteAndRead(t *testing.T) {
+	db := easypg.ConnectForTest(t, emptyMigration())
+	defer db.Close()
+
+	store := newTestSQLBackingStoreWithDB(t, db, SQLBackingStoreOpts{
+		BatchSize: 10,
+		MaxEvents: 100,
+	})
+	defer store.Close()
+
+	// Write events
+	mustWriteSQL(t, store, testEvent("event-1"))
+	mustWriteSQL(t, store, testEvent("event-2"))
+	mustWriteSQL(t, store, testEvent("event-3"))
+
+	// Read batch (should get all events in FIFO order)
+	events := mustReadBatchSQL(t, store)
+	assert.DeepEqual(t, "event count", len(events), 3)
+	assert.DeepEqual(t, "event 1 ID", events[0].ID, "event-1")
+	assert.DeepEqual(t, "event 2 ID", events[1].ID, "event-2")
+	assert.DeepEqual(t, "event 3 ID", events[2].ID, "event-3")
```

Comment:
For comparable values like this one (i.e. values that Go allows to compare with `==`), `assert.Equal` is to be prefered because it does not require reflection.